Source/Domain/Kysect.Tamgly.Core/Graphs/GraphNode.cs

```diff
@@ -0,0 +1,51 @@
+﻿namespace Kysect.Tamgly.Core.Graphs;
+
+public class GraphNode<T>
+{
+    public Guid Id { get; set; }
+    public T Value { get; }
+    public IReadOnlyCollection<GraphNode<T>> DirectChild { get; set; }
+
+    public GraphNode(Guid id, T value, IReadOnlyCollection<GraphNode<T>> directChild)
+    {
+        ArgumentNullException.ThrowIfNull(directChild);
+
+        Id = id;
+        Value = value;
+        DirectChild = directChild;
+    }
+
+    public IReadOnlyCollection<GraphPath<T>> EnumeratePathToLeaves()
+    {
+        if (!DirectChild.Any())
+        {
+            return new[] { GraphPath<T>.Empty.AppendToStart(Value) };
+        }
+
+        List<GraphPath<T>> result = new List<GraphPath<T>>();
+        foreach (GraphNode<T> child in DirectChild)
+        {
+            foreach (GraphPath<T> pathToLeaf in child.EnumeratePathToLeaves())
+            {
+                result.Add(pathToLeaf.AppendToStart(child.Value));
+            }
+        }
+
+        return result;
+    }
```

Comment:
Очень странный метод.
- Непонятный нейминг.
- Отсутствие доки.
- В одном месте мы возвращаем значение текущего узла, в другом нет.

Выглядит будто было проиграно в рекурсию, мне кажется что подразумевалось что-то такое: 
```suggestion
    public IReadOnlyCollection<GraphPath<T>> EnumeratePathToLeaves()
    {
        if (!DirectChild.Any())
        {
            return new[] { GraphPath<T>.Empty.AppendToStart(Value) };
        }
        
        return DirectChild
            .SelectMany(c => c.EnumeratePathToLeaves())
            .Select(p => p.AppendToStart(Value))
            .ToList();
    }
```

(Если нет, то скорее всего где-то есть костыли)

Reply:
Ну и оно не используется -_-

Reply:
Да, действительно была ошибки и там нужно Value подставлять. Метод действительно не очевидный без юзаджей. Это я проиграл в итерации и начал делать реализацию на будущее, чтобы более оптимально выбираться задачу. Думаю, лучше пока удалить т.к. не будет использоваться.